digicampipe/scripts/pulse_shape.py

```diff
@@ -1,78 +1,121 @@
+    charge_min = np.min(charge_range)
+    charge_max = np.max(charge_range)
+    integration_min = np.min(integration_range)
+    integration_max = np.max(integration_range)
     events = calibration_event_stream(input_files)
-    Rough_factor_between_single_pe_amplitude_and_integral = 21 / 5.8
+    events = fill_digicam_baseline(events)
+    if "SST1M_01_201805" in input_files[0]:  # fix data in May
+        print("WARNING: correction of the baselines applied.")
+        events = correct_wrong_baseline(events)
+    events = subtract_baseline(events)
     histo = None
-
-    for e in tqdm(events):
+    n_sample = 0
+    n_pixel = 0
+    for e in events:
         adc = e.data.adc_samples
-        adc = adc - e.data.digicam_baseline[:, None]
-
-        # I just integrate between sample 10 and 30 to normalize a bit
-        # normalizing to maximum_amplitude = 1 is too "sharp"
-        integral = adc[:, 10:30].sum(axis=1)
-
-        # handling special case .. we say negative integrals make no sense
-        # and zero integral simply means there was no pulse at all.
-        # so we clip at 1
-        integral = integral.clip(1)
-
-        adc = (
-                  adc / integral[:, None]
-              ) * Rough_factor_between_single_pe_amplitude_and_integral
-
-        arrival_time_in_ns = estimate_arrival_time(adc) * 4
-        time_in_ns = np.arange(adc.shape[1]) * 4
-
-        # TODO: Would be nice to move this out of the loop
+        integral = adc[:, slice(integration_min, integration_max)].sum(axis=1)
+        adc_norm = adc / integral[:, None]
+        arrival_time_in_ns = estimate_time_from_leading_edge(adc) * 4
         if histo is None:
+            n_pixel, n_sample = adc_norm.shape
             histo = Histogram2dChunked(
-                shape=(adc.shape[0], 101, 101),
-                range=[[-10, 40], [-0.2, 1.5]]
+                shape=(n_pixel, n_bin, n_bin),
+                range=[time_range_ns, amplitude_range]
             )
-
+        else:
+            assert adc_norm.shape[0] == n_pixel
+            assert adc_norm.shape[1] == n_sample
+        time_in_ns = np.arange(n_sample) * 4
+        bad_charge = np.logical_or(
+            integral < charge_min,
+            integral > charge_max
+        )
+        arrival_time_in_ns[bad_charge] = -np.inf  # ignored by histo
         histo.fill(
             x=time_in_ns[None, :] - arrival_time_in_ns[:, None],
-            y=adc
+            y=adc_norm
         )
-
-    outfile = h5py.File(outfile_path)
-    dset = outfile.create_dataset(
-        name='adc_count_histo',
-        data=histo.contents(),
-        compression='gzip'
-    )
-    dset.attrs['extent'] = histo.extent
+    histo.save(output_hist)
+    print('2D histogram of pulse shape for all pixel saved as', output_hist)
 
 
 def entry():
     args = docopt(__doc__)
-    if args['--output'] is None:
-        args['--output'] = args['<input_files>'][0] + '.h5'
+    inputs = args['<input_files>']
```

Comment:
I think at some point we should come with a class of something to handle command line arguments for all kind of scripts. 
We should also try to "harmonize" options / arguments names

Reply:
good idea, I opened an issue #261